scripts/bootstrap.sh.in

```diff
@@ -144,56 +144,77 @@ retry_dns_count: 3
 EOF
 }
 
-set_salt_command() {
-    echo "Setting salt master command"
-    # shellcheck disable=SC2046
-    SALT_MASTER_CALL="crictl exec -i $(crictl ps -q --label io.kubernetes.container.name=salt-master --label io.kubernetes.pod.namespace=kube-system)"
+get_salt_container() {
+    local salt_container
+    salt_container=''
+    while [ -z "$salt_container" ]; do
+        salt_container="$(crictl ps -q \
+            --label io.kubernetes.pod.namespace=kube-system \
+            --label io.kubernetes.container.name=salt-master \
+            --state Running)"
+    done
+    echo "$salt_container"
 }
 
 sync_salt() {
-    $SALT_MASTER_CALL salt '*' saltutil.sync_all refresh=True \
+    local salt_master_call
+    salt_master_call="crictl exec -i $(get_salt_container)"
+
+    $salt_master_call salt '*' saltutil.sync_all refresh=True \
+        saltenv=metalk8s-@@VERSION
+
+    $salt_master_call salt '*' mine.update
+
+    $salt_master_call salt-run saltutil.sync_all saltenv=metalk8s-@@VERSION
+    $salt_master_call salt-run saltutil.sync_roster saltenv=metalk8s-@@VERSION
+    $salt_master_call salt-run metalk8s_saltutil.sync_auth \
         saltenv=metalk8s-@@VERSION
-    $SALT_MASTER_CALL salt '*' mine.update
-    local SALT_CONTAINER
-    SALT_CONTAINER="$(crictl ps -q \
-            --label io.kubernetes.pod.namespace=kube-system \
-            --label io.kubernetes.container.name=salt-master \
-            --state Running)"
-    test -n "$SALT_CONTAINER"
-    crictl exec -i "$SALT_CONTAINER" \
-        salt-run saltutil.sync_all saltenv=metalk8s-@@VERSION
-    crictl exec -i "$SALT_CONTAINER" \
-        salt-run saltutil.sync_roster saltenv=metalk8s-@@VERSION
-    crictl exec -i "$SALT_CONTAINER" \
-        salt-run metalk8s_saltutil.sync_auth saltenv=metalk8s-@@VERSION
 }
 
 orchestrate_bootstrap() {
-    # TODO: FIX register IP
-    local cidr
-    local registry_ip
-    cidr=$(salt-call --local pillar.get networks:control_plane --out=txt | cut -f 2 -d ' ')
-    registry_ip=$(salt-call --local network.ip_addrs cidr="$cidr" --out=txt | sed "s/.*'\(\([0-9]\{1,3\}\.\)\{3\}[0-9]\{1,3\}\)'.*/\1/")
     # Grains must be set (in `/etc/salt/grains`) *before* invoking `salt-call`,
     # otherwise grains set during execution won't be taken into account
     # properly.
     ${SALT_CALL} --local --retcode-passthrough state.sls \
         metalk8s.node.grains \
         saltenv=metalk8s-@@VERSION \
         pillarenv=metalk8s-@@VERSION
+
+    
+    # TODO: Registry IP should not need a pillar override here
+    local cidr
+    local registry_ip
+    cidr=$( \
```

Comment:
Could your branch be out of date?

Reply:
Probably! Will rebase.

Reply:
So I had no conflict when rebasing, did you expect this to retrieve the IP using grains, as introduced in 37340e4? I'm fine with adding it.

Reply:
I must have had another PR which isn't merged yet in mind.